crates/cli/cli/src/commands/clear/packets.rs

```diff
@@ -0,0 +1,128 @@
+use ibc_relayer_types::core::ics24_host::identifier::ChannelId;
+use ibc_relayer_types::core::ics24_host::identifier::ClientId;
+use ibc_relayer_types::core::ics24_host::identifier::PortId;
+
+use crate::Result;
+
+#[derive(Debug, clap::Parser)]
+pub struct PacketsClear {
+    #[clap(
+        long = "chain",
+        required = true,
+        value_name = "CHAIN_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chain to query"
+    )]
+    chain_id: ChainId,
+
+    #[clap(
+        long = "port",
+        required = true,
+        value_name = "PORT_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the port"
+    )]
+    port_id: PortId,
+
+    #[clap(
+        long = "channel",
+        alias = "chan",
+        required = true,
+        value_name = "CHANNEL_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the channel"
+    )]
+    channel_id: ChannelId,
+
+    #[clap(
+        long = "client",
+        required = true,
+        value_name = "CLIENT_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the client"
+    )]
+    client_id: ClientId,
+
+    #[clap(
+        long = "counterparty-chain",
+        required = true,
+        value_name = "COUNTERPARTY_CHAIN_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the counterparty chain to query"
+    )]
+    counterparty_chain_id: ChainId,
+
+    #[clap(
+        long = "counterparty-client",
+        required = true,
+        value_name = "COUNTERPARTY_CLIENT_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the counterparty client"
+    )]
+    counterparty_client_id: ClientId,
+
+    #[clap(
+        long = "counterparty-port",
+        required = true,
+        value_name = "COUNTERPARTY_PORT_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the counterparty port"
+    )]
+    counterparty_port_id: PortId,
+
+    #[clap(
+        long = "counterparty-channel",
+        required = true,
+        value_name = "COUNTERPARTY_CHANNEL_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the counterparty channel"
+    )]
+    counterparty_channel_id: ChannelId,
+}
+
+impl Runnable for PacketsClear {
+    async fn run(&self, builder: CosmosBuilder) -> Result<Output> {
+        let chain = builder.build_chain(&self.chain_id).await?;
+
+        let relayer = builder
+            .build_relay(
+                RelayAToBTarget,
+                &self.chain_id,
+                &self.counterparty_chain_id,
+                &self.client_id,
+                &self.counterparty_client_id, // nothing to pass here
+            )
+            .await?;
+
+        let target_height = chain.query_chain_height().await?;
+
+        relayer
+            .send_update_client_messages(DestinationTarget, &target_height)
+            .await
+            .wrap_err("Failed to send update client message")?;
```

Comment:
It seems that the `clear_packets` method doesn't prepend a client update

Reply:
That's fine for now but eventually it would nice if `clear_packets` could determine whether or not a client update is needed and send it itself.

Reply:
Is this still the case? The relay context should already automatically send update client message, so not sure why it is not working here.

Reply:
It might have been caused by the partial clear implementation. What is odd is that updating the client at that time seemed to work around the issue.
I tried removing the update client now that the full clear has been merged and it seems to be unnecessary. I will remove it.

Reply:
3e5f94bb308764733121082de6b9bc68e0a56657